### ISSUE TITLE
v3 editor report an error while download (The blocks could work in v0)

### ISSUE THREAD

**Issue by jxgx072037** [Microsoft/pxt-microbit#1261]
**1. Desktop :**
 - OS: Windows 10/Mac
 - Browser: Chrome 68+
 - Version: pxt-microbit, v3, https://makecode.microbit.org/beta#editor

**2. Describe the bug**

I add the minode package(https://github.com/jxgx072037/pxt-minode) into pxt since I want to get the dht11 tempreture.

I got an error after I clicked the download button. It report 'compliation failed, please check your code for errors.' The blocks is very simple, just show the dht11 tempreture value.

![default](https://user-images.githubusercontent.com/4236328/45643181-0513cf80-baed-11e8-8cc1-d213b34f2cff.png)

**The blocks could work in pxt v0 editor.**

2.1. The minode package which I used: https://github.com/jxgx072037/pxt-minode (I just folked it from Minode and change nothing)
2.2. I followed the instruction (https://makecode.com/blog/github-packages) and I found that the function DHTGetTemperature in minode.ts report an error just like below, 'expecting non-number at parameter 1 of DHTGetTemperature(...)(shim=minode: :DHTGetTemperature) / I__'

![wx20180918-023733](https://user-images.githubusercontent.com/4236328/45643452-b4e93d00-baed-11e8-8dc6-cc4a90e46252.png)

In the enum.ts, the ConnName just like this,

![wx20180918-025231](https://user-images.githubusercontent.com/4236328/45644062-3ee5d580-baef-11e8-945e-9806eae32fdc.png)

How could I fix this?

Thanks!!!

**Comment by pelikhan** [comment]
I can't seem to repro this issue with the official minode package. Please use the share feature to share your program.

**Comment by pelikhan** [comment]
JS compilation fails but .hex file compilation fails silently.

**Comment by riknoll** [comment]
Possibly related to https://github.com/Microsoft/pxt-microbit/issues/1019

**Comment by jxgx072037** [comment]
Hi @riknoll ,

May I ask when would next version release which could solve this issue?

**Comment by riknoll** [comment]
@jxgx072037 next-release is just a tag we use internally for issues that we want to resolve before the next release whenever that might be. We don't have a release date to give you, unfortunately.

**Comment by riknoll** [comment]
Okay, I think I figured out the issue. When we parse C++ files we keep track of the enums we see and check function declarations to make sure the enum exists. Our C++ parser doesn't support the enum declaration syntax used by minode (see [here](https://github.com/minodekit/pxt-minode/blob/master/libs/MiNodeConn.h#L8)). They got around that in v0 by checking in a file with the enums re-declared in TypeScript ([here](https://github.com/minodekit/pxt-minode/blob/master/enums.ts)). Our old v0 compiler didn't bother to do much type checking of function arguments in C++ so it ignored this issue and assumed everything passed. Our new compiler is much stricter and actually checks the type of each argument so it complains when the enum is missing.

So, there are two options:
1. Fix the package to use the different enum syntax and remove the enums.ts file that I linked above
2. Fix our parser to handle the `typedef enum {...}whatever;` syntax and remove the enums.ts file that I linked above.

I don't think we can simply pass the TS info to the C++ parser because we wouldn't be able to run the TypeScript compiler on the project before we parse the C++ code. We'd have to parse the ts files ourselves which I don't think we should do.

Anyway, passing this one off to @abchatra who knows much more about C++ than I.
